connection.go

```diff
@@ -0,0 +1,68 @@
+package mgod
+
+import (
+	"context"
+	"time"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var dbConn *mongo.Database
+var defaultTimeout = 10 * time.Second
+
+// ConnectionConfig is the configuration for a MongoDB connection.
+type ConnectionConfig struct {
+	// Timeout is the timeout for the connection.
+	Timeout time.Duration
+}
+
+// SetDefaultConnection sets the default connection to be used by the package.
+func SetDefaultConnection(conn *mongo.Database) {
+	dbConn = conn
+}
+
+// ConfigureDefaultConnection opens a new connection using the provided config options and sets it as a default connection to be used by the package.
+func ConfigureDefaultConnection(config *ConnectionConfig, dbName string, opts ...*options.ClientOptions) error {
+	if config == nil {
+		config = defaultConnectionConfig()
+	}
+
+	client, err := newClient(config, opts...)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), config.Timeout)
+	defer cancel()
```

Comment:
This ctx is used while pinging the MongoDB server to check if the connection is properly established. The main reason to do this is to prevent combining two operations i.e. connect and ping in the same timeout context. Let me know if we want to create a separate function to get the timeout context.